osu.Game.Rulesets.Osu/Skinning/Default/SpinnerRotationTracker.cs

```diff
@@ -77,6 +80,10 @@ protected override void Update()
             Rotation = (float)Interpolation.Damp(Rotation, currentRotation, 0.99, Math.Abs(Time.Elapsed));
         }
 
+        private float currentDelta;
+        private int currentDirection;
+        private readonly List<(double time, float delta)> turningPoints = new List<(double, float)>();
```

Comment:
This isn't being cleared in `resetState()` but should be.

Failure to do this manifests on maps with several spinners, where due to pooling and instance reuse, clearing one spinner will cause every subsequent spinner on the map to pretty much auto-clear.

Reply:
Also makes me wonder if this will even do what it's intending to do when fixed. Wouldn't it break rewinds? Does this data need to be stored somewhere persistent (non-pooled) per spinner?

Reply:
Oh yeah good point. This needs to be stored in the judgement to be 100% correct.